codespell_lib/_codespell.py

```diff
@@ -1021,7 +1022,14 @@ def _script_main() -> int:
 
 def main(*args: str) -> int:
     """Contains flow control"""
-    options, parser, used_cfg_files = parse_options(args)
+    try:
+        options, parser, used_cfg_files = parse_options(args)
+    except configparser.Error as e:
+        print(
+            f"ERROR: ill-formed config file: {e.message}",
+            file=sys.stderr,
+        )
+        return EX_CONFIG
```

Comment:
I think this will mean you can't ever use Codespell if your setup.cfg has an invalid config for example (as you can't override just append these):
https://github.com/codespell-project/codespell/blob/ca44052fa85a02609492e6b51771a058f79844e2/codespell_lib/_codespell.py#L547-L548

Not sure what the best solution is, but thought it's worth flagging before we merge...

Reply:
Maybe warn if it's the root-level `setup.cfg` and error otherwise?

Reply:
That was already the case, with an exception raised. Nobody has complained.

Reply:
> That was already the case, with an exception raised. Nobody has complained.

Oops, I didn't think of that! :facepalm: